### PR TITLE
Added feature + Bug fix

### PR DESCRIPTION
**Pull request by maximus-sallam**
Added feature: Global npm installation is restricted on managed hosting platforms due to security considerations. As an employee at a shared hosting company, I have encountered numerous instances where we had to inform customers that Magepack is not supported on our platform due to the mandatory global installation requirement. To address this challenge, I have implemented code modifications and supplied the necessary shared object files, enabling local execution for improved security and compatibility.

Bug fix: The feature possibly introduced a bug that caused collectModules.js to become unresponsive during the product.js stage of generation, specifically at the rjsResolver step. The issue was with the use of a Promise in the code. Promises in JavaScript are used for asynchronous operations, indicating that a function will complete at some point in the future. In this case, the Promise was used to wait for the rjsResolver operation to finish before continuing with the code execution.

The problem with this approach was that it caused the code to pause and wait for the rjsResolver operation to complete before moving on to the next steps. As a result, the code appeared to freeze or stall at that point because it was waiting for the Promise to resolve.

To address this, I replaced the Promise with a callback function named onRjsResolverComplete. By using this callback, I ensured that the code would continue running without waiting for the rjsResolver operation to finish. This change allowed the code to proceed smoothly without any interruptions, and it no longer appeared to freeze at that stage of execution.

### CODE REVIEW

**Review by maximus-sallam — COMMENTED** [review]
Please consider applying my updates to master.

File: lib/generate/collector/product.js [hunk 1]
```diff
@@ -1,20 +1,20 @@
-const merge = require('lodash.merge');
+const merge = require(`lodash.merge`);
```
Comment (fredden) [inline lib/generate/collector/product.js:1]: What's the reason for this change from quotes to ticks?

File: lib/generate/collector/product.js [hunk 2]
```diff
@@ -28,21 +28,22 @@ const product = async (
 
     const bundleName = bundleConfig.name;
 
-    logger.info(`Collecting modules for bundle "${bundleName}".`);
+    logger.info(`Collecting modules for bundle ${bundleName}`);
```
Comment (fredden) [inline lib/generate/collector/product.js:31]: Why remove the quotes around `bundleName`?

File: lib/generate/collectModules.js [hunk 3]
```diff
@@ -14,14 +14,11 @@ module.exports = async (page) => {
...
-                require(['rjsResolver'], (resolver) => {
-                    resolver(() => resolve());
-                });
-            })
-    );
+    await page.evaluate(() => {
+        require(['rjsResolver'], (resolver) => {
+            resolver(() => {});
+        });
+    });
```
Comment (fredden) [inline lib/generate/collectModules.js:21]: Can you confirm that this will still wait for require.js to complete? We need to wait until the page has finished loading before we continue. It looks like this change removes the "wait" part of this step.

File: README.md [hunk 4]
```diff
@@ -24,18 +24,38 @@ Here are the requirements for Magepack to work:
...
+npm install magepack
+```
+The following commands allow you to download the essential x86_64 shared object files required by Chrome to operate in headless mode.
+
+Note: These files are designed specifically for x86_64 architecture.
+- I can provide the shared object files for other architectures upon request.
+
+```shell
+mkdir -p $HOME/.local/lib
+wget https://github.com/maximus-sallam/magepack/raw/master/x86_64/lib.tar.gz
```
Comment (fredden) [inline README.md]: If you want these changes to be accepted here, you may like to update the URL to be *this* project instead of your fork thereof.

File: lib/generate/collector/product.js (hunk 1 above)
Comment (maximus-sallam) [reply lib/generate/collector/product.js:1]: Hello Fredden. Thank you for your question. This was changed during my debugging. I had added in several lines to print to console the status and progress of processes at different points in the code, including API requests and responses. During this time, my print statements were displaying the variable name (e.g. ${some_variable}) I used, rather than the value of the variable as I had intended. both single quotes and double quotes caused the console print to display the variable name instead of the variable value, but using backticks enabled the variable value to be displayed.

File: lib/generate/collector/product.js (hunk 2 above)
Comment (maximus-sallam) [reply lib/generate/collector/product.js:31]: Hello Fredden. Same as with the use of backticks previously, I had changed the quotes to backticks across the entire file for uniformity. As for the quotes around the variable, this was due to my uncertainty of whether the variable would display the name or value if enclosed in quotes. I believe I may have been confusing the rules in Python with JavaScript rules.

File: lib/generate/collector/product.js (hunk 1 above)
Comment (fredden) [reply lib/generate/collector/product.js:1]: Thanks for confirming that this is not an intended change. Please can you adjust this so that the changes in this pull request only show what you intended to change.

File: lib/generate/collectModules.js (hunk 3 above)
Comment (maximus-sallam) [reply lib/generate/collectModules.js:21]: The new code does not maintain the "wait" functionality as the original code. The original code waits for the rjsResolver to signal completion, whereas the new code only waits for the require function to execute, not for all modules to load. However, during my testing, this is where the rjsResolver would hang. I believe you may have ran into a similar issue previously with rjsResolver hanging, according to the comments in blockMagepack.js. I think I can improve upon this. I'll make changes and perform some tests and submit another PR once I've debugged the rjsResolver hanging issue properly. My goal is to not have to block assets due to them causing rjsResolver to hang.

File: README.md (hunk 4 above)
Comment (maximus-sallam) [reply README.md]: My apologies. I'll fix this now.

Comment (maximus-sallam) [reply README.md]: I have updated the URL to reflect your repo, however, since you are not hosting the shared object files, this URL will return a 404. I should note that these shared object files were the remaining files needed to get Chrome operational within our infrastructure, and not the total shared objects required. The files I've provided do not account for the required shared objects that were already installed within our infrastructure. For a total list of required shared object files, see the Dependencies list here: https://pptr.dev/troubleshooting
